clab/config/transport.go

```diff
@@ -0,0 +1,73 @@
+package config
+
+import (
+	"fmt"
+	"strings"
+)
+
+type Transport interface {
+	// Connect to the target host
+	Connect(host string) error
+	// Execute some config
+	Write(snip *ConfigSnippet) error
```

Comment:
I think `Write()` shouldn't know about `ConfigSnippet`, it needs to write raw byte slice instead

Reply:
Let me have a look at this again, but believe from simple debugging etc the code currently uses much more than just the byte slice

Reply:
yes, I see the reasoning behind propagating a richer struct as an input, but imho that makes writing transports harder as we drag the strcut ConfigSnippet to a transport which shouldn't really know anything about it

by making the transport work on primitive types like `[]byte` I believe we will have a cleaner code and distinctive packages separation.